sim/battle.ts

```diff
@@ -6,6 +6,7 @@
  */
 import Dex = require('./dex');
 global.toId = Dex.getId;
+import {MoveEffectiveness} from './constants';
```

Comment:
I still don't like having an external constants file. We've always just declared constants at the place they're used. Especially with string enums, you can just use the strings themselves instead of needing to import types.

Reply:
> type MoveEffectiveness = 'invalid' | 'neutral' | 'immune' | 'resist' | 'weak';

Didn't you want type-checking? ``enum``s are the ideal way for that :|

Reply:
Type literals in Typescript accomplish the same thing, and are the pattern we've been using throughout the codebase